book-1-orientation/chapters/CLASSES_08-DUCK_TYPING.md

```diff
@@ -0,0 +1,78 @@
+# Critters and Croquettes: Day 8
+
+>You bolt awake, covered in a cold sweat. As you sigh in relief that it was all just a dream, you realize something almost as scary: it could really happen! Just yesterday you were testing your code and you put a goose in the petting zoo without a second thought. But geese belong in Critter Cove, not Varmint Village. Right now, there's no limit to the type of animals that get placed in an attraction. 
+>
+>In Python, lists can contain any combination of object types. A single list could contain an integer, a boolean, and a string. This means that the `animals` attribute of your attractions classes can contain **any** critter type, regardless of whether it belongs there.
+>
+>You decide that you'll need to add a check to the `addAnimal` method of your attractions classes to make sure a critter belongs in the attraction you're adding it to. Luckily, you already have a built-in way of determining what kind of critter you're dealing with...
+
+### Using Your Inherited Class Properties to Restrict Lists
+
+You are going to see two ways to make sure that only walking animals can be added to a petting zoo. One is by doing type checking, which is done in statically typed languages like Java or C#, and the other is the Pythonic way.
+
+> #### `attractions/petting_zoo.py`
+
+```py
+from . import Attraction
+from movements import Swimming
+
+class PettingZoo(Attraction):
+
+    def __init__(self, name, description):
+        super().__init__(name, description)
+    
+    # Number 1: Duck typing check
+    def add_animal_pythonic(self, animal):
+        try:
+            if animal.swim_speed > -1:
+                self.animals.append(animal)
+        except AttributeError as ex:
+            print(f'{animal} doesn\'t like to be petted, so please do not put it in the {self.name} attraction.')
+
+    # Number 2: Actual typing check
+    def add_animal_type_check(self, animal):
+        if isinstance(animal, Swimming):
+            self.animals.add(animal)
+        else:
+            print(f'{animal} doesn\'t like to be petted, so please do not try to put it in the {self.name} attraction.')
+```
+
+> ### Sidebar: Duck Typing
+>
+> "If it looks like a duck, sounds like a duck, acts like a duck, and smells like a duck, then it can do all the things we want a duck to do. It's a duck."
+>
+> The idea is that it doesn't actually matter what type my data is - just whether or not I can do what I want with it.
+
```

Comment:
I feel like this is a bit confusing. Because you say there's type checking, which is done in other languages. And there's the Pythonic way. But then in the example, you introduce duck typing first. And it's never made clear that duck typing is the Pythonic way. Maybe say here is type checking. This is what is done in other languages BUT we will not be doing that in Python. Now here is duck typing, which is the Pythonic way. This also stems from the fact that a lot of students will go into Keahua assuming you can do either or both. And then we look at their code when they think they are at MVP and have to be like "Go implement duck typing!"